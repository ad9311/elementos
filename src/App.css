/* App.css */

@import url('https://fonts.googleapis.com/css2?family=Titillium+Web:wght@400;600;700&display=swap');

:root {
  --light-blue: #3C83B9;
  --medium-blue: #2F6690;
  --dark-blue: #19364D;
  --font-color: #FBFFF1;
}

body {
  margin: 0;
  background-color: var(--light-blue);
  color: var(--font-color);
  font-family: 'Titillium Web', sans-serif;
}

h1, h2, h3, h4, p {
  margin: 0;
}

a:hover {
  text-decoration: none;
}
