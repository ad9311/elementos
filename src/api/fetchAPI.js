const fetchAPI = async () => {
  const URL = 'https://periodic-table-api.herokuapp.com/';
  const request = await fetch(URL, {
    method: 'GET',
  });
  const response = await request.json();
  return response;
};

export default fetchAPI;
