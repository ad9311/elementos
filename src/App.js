import React from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import Header from './components/header/Header';
import Home from './components/pages/Home';
import Details from './components/pages/Details';
import './App.css';

const App = () => (
  <Router basename={process.env.PUBLIC_URL}>
    <Header />
    <main>
      <Switch>
        <Route exact path="/">
          <Home />
        </Route>
        <Route path="/details/">
          <Details />
        </Route>
      </Switch>
    </main>
  </Router>
);

export default App;
