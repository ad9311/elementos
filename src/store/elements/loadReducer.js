import { createAsyncThunk } from '@reduxjs/toolkit';
import fetchAPI from '../../api/fetchAPI';

const FETCH_ELEMENTS = 'periodic_table/FETCH_ELEMENTS/fulfilled';
const REARRANGE_ELEMENTS = 'periodic_table/REARRANGE_ELEMENTS/fulfilled';

const initialState = {
  status: 'default',
  elements: [],
};

export const fetchElements = createAsyncThunk('periodic_table/FETCH_ELEMENTS', async () => {
  const elements = await fetchAPI();
  return elements;
});

const loadReducer = (state = initialState, action) => {
  switch (action.type) {
    case FETCH_ELEMENTS:
      return {
        status: 'ready',
        elements: [...action.payload],
      };
    case REARRANGE_ELEMENTS:
      return {
        status: 'rearranged',
        elements: [...action.payload],
      };
    default:
      return state;
  }
};

export default loadReducer;
