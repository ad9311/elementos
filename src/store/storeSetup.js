import { createStore, combineReducers, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import loadReducer from './elements/loadReducer';
import inspectReducer from './elements/inspectReducer';
import arrangeReducer from './elements/arrangeReducer';

const combineMiddleware = [thunk];

const reducer = combineReducers(
  {
    periodicTable: loadReducer,
    inspect: inspectReducer,
    arrange: arrangeReducer,
  },
);

const store = createStore(
  reducer,
  applyMiddleware(...combineMiddleware),
);

export default store;
