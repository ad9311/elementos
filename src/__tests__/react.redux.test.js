import { render, screen, fireEvent } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../store/storeSetup';
import loadReducer from '../store/elements/loadReducer';
import fetchAPI from '../api/fetchAPI';
import App from '../App';

describe('Load Table reducer', () => {
  test('initial state', () => {
    expect(loadReducer(undefined, {})).toEqual(
      {
        status: 'default',
        elements: [],
      },
    );
  });

  test('Fetch API', async () => {
    const elements = await fetchAPI();
    expect(elements[0].name).toEqual('Hydrogen');
  });

  test('App', async () => {
    render(
      <Provider store={store}>
        <App />
      </Provider>,
    );
    expect(await screen.findByText('Elementos')).toMatchSnapshot();
  });

  it('Home', async () => {
    render(
      <Provider store={store}>
        <App />
      </Provider>,
    );
    expect(await screen.findByText('Carbon')).toMatchSnapshot();
  });

  it('Details page for Carbon', async () => {
    render(
      <Provider store={store}>
        <App />
      </Provider>,
    );
    fireEvent.click(await screen.findByText('Carbon'));
    expect(screen.findByText('4300 K')).toMatchSnapshot();
  });
});
