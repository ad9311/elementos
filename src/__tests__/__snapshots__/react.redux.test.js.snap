// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Load Table reducer App 1`] = `
<h4>
  Elementos
</h4>
`;

exports[`Load Table reducer Details page for Carbon 1`] = `Promise {}`;

exports[`Load Table reducer Home 1`] = `
<h3>
  Carbon
</h3>
`;
